angularjs/angular.d.ts

```diff
@@ -1413,13 +1402,15 @@ declare module angular {
         (data: any, headersGetter: IHttpHeadersGetter, status: number): any;
     }
 
+    type HttpHeaderType = {[requestType: string]:string|((config:IRequestConfig) => string)};
+
     interface IHttpRequestConfigHeaders {
-        [requestType: string]: string|(() => string);
-        common?: string|(() => string);
-        get?: string|(() => string);
-        post?: string|(() => string);
-        put?: string|(() => string);
-        patch?: string|(() => string);
+        [requestType: string]: any;
+        common?: any;
+        get?: any;
+        post?: any;
+        put?: any;
+        patch?: any;
```

Comment:
It's a result of discussion in issue #5855. The problem is `IHttpRequestConfigHeaders` interface is used in two different places:
> Actually, there are two headers, one in $http[Provider].defaults and one in IRequestConfig - an object that can be passed to $http on each call.
> They have different types: the one in IRequestConfig is an object hash of header name to header content, whereas the one in defaults is a hash from request method to the hash above.

I suggest `any` as a temporary solution until this issue is resolved.

https://github.com/DefinitelyTyped/DefinitelyTyped/issues/5855
https://github.com/DefinitelyTyped/DefinitelyTyped/commit/a42706711d711cf12ec26b8bb5f08520fc948253#commitcomment-13395727